src/resolve_links.cpp

```diff
@@ -69,6 +69,19 @@ void resolve_links::match(std::shared_ptr<const torrent_info> const& ti
 	// only torrents with the same piece size
 	if (ti->piece_length() != m_torrent_file->piece_length()) return;
```

Comment:
Since v2 torrents use merkle hash trees that always terminate at the block level (16 kiB), the piece size doesn't actually have to match! So, for v2 torrents, you can check matches before this check